docs/recipes/low_freq_proc.qmd

```diff
@@ -0,0 +1,183 @@
+
+# Get a spool to work on
+sp = dc.get_example_spool().update()
+
+# Sort the spool
+sp = sp.sort("time")
+# You may also want to sub-select the spool for the desired distance or time samples before proceeding.
+
+# Get the first patch
+pa = sp[0]
+
+# Define the target sampling interval (in sec.)
+dt = 10 
+# With a sampling interval of 10 seconds, the cutoff frequency (Nyquist frequency) is determined to be 0.05 Hz 
+cutoff_freq = 1 / (2*dt)
+
+# Safety factor for the low-pass filter to avoid ailiasing 
+filter_safety_factor = 0.9
+
+# Enter memory size to be dedicated for processing (in MB)
+memory_limit_MB = 10_000
+
+# Define a tolerance for determining edge effects (used in next step)
+tolerance = 1e-3
+```
+
+## Calculate chunk size and determine edge effects
+
+To chunk the spool, first we need to figure out the chunk size based on machine's memory size so we ensure we can load and process patches with no memory issues. Longer chunk size (longer patches) increases computation efficiency. 
+
+Notes: 
+
+1. The `processing_factor` is required because certain processing routines involve making copies of the data during the processing steps. It should be determined by performing memory profiling on an example dataset for the specific processing routine. For instance, the combination of low-pass filtering and interpolation, discussed in the next section, requires a processing factor of approximately 5.
+2. The `memory_safety_factor` is optional and helps prevent getting too close to the memory limit.
+
+```{python}
+# Get patch's number of bytes per seconds (based on patch's data type) 
+pa_bytes_per_second = pa.data.nbytes / pa.seconds
+# Define processing factor and safety factor 
+processing_factor = 5  
+memory_safety_factor = 1.2 
+
+# Calculate memory size required for each second of data to get processed
+memory_size_per_second = pa_bytes_per_second * processing_factor * memory_safety_factor
+memory_size_per_second_MB = memory_size_per_second / 1e6
+
+# Calculate chunk size that can be loaded (in seconds)
+chunk_size = memory_limit_MB / memory_size_per_second_MB 
+```
+
+Next, we need to determine the extent of artifacts introduced by low-pass filtering at the edges of each patch. To achieve this, we apply LF processing to a delta function patch, which contains a unit value at the center and zeros elsewhere. The distorted edges are then identified based on a defined threshold.
+
+```{python}
+# Retrieve a patch of appropriate size for LF processing that fits into memory
+pa_chunked_sp = sp.chunk(time=chunk_size)[0] 
+# Create a delta patch based on new patch size
+delta_pa = dc.get_example_patch("delta_patch", dim="time", patch=pa_chunked_sp)
+
+# Apply the low-pass filter on the delta patch
+delta_pa_low_passed = delta_pa.pass_filter(time=(None, cutoff_freq * filter_safety_factor))
+# Resample the low-passed filter patch
+new_time_ax = np.arange(delta_pa.attrs["time_min"], delta_pa.attrs["time_max"], np.timedelta64(dt, "s"))
```

Comment:
Can you just use `patch.resample` here?

Reply:
Apparently, they do not provide the same results:
```python
import numpy as np
import dascore as dc

pa = dc.get_example_patch()
dt = 5 

resampled_pa = pa.resample(time=np.timedelta64(dt, 's')) # has one time sample at zero

new_time_ax = np.arange(pa.attrs["time_min"], pa.attrs["time_max"], np.timedelta64(dt, "s"))
interpolated_pa = pa.interpolate(time=new_time_ax) # has two time samples at 0 and 5 sec
```

Reply:
Ah, that is odd. I think #10 might be related.

Reply:
I haven't worked with `patch.resample()` much; do you think this behavior is expected?

Reply:
In both cases they are just calling underlying scipy functions so I don't think there is much we can do about it anyway.